pwiz_tools/Skyline/Model/AuditLog/AuditLogEntry.cs

```diff
@@ -41,6 +40,63 @@
 
 namespace pwiz.Skyline.Model.AuditLog
 {
+
+    public class AuditLogException : Exception
+    {
+        public AuditLogException(string pMessage) : base(pMessage)
+        {
+        }
+        public AuditLogException(string pMessage, Exception pInner) : base(pMessage, pInner)
+        {
+        }
+
+        /// <summary>
+        /// Checks if the given exception was thrown due to problems with audit log processing
+        /// </summary>
+        /// <param name="ex">Exception to check</param>
+        /// <returns>True if this or any nested exception is of <see cref="AuditLogException"> AuditLogException </see> type.</returns>
+        public static bool IsAuditLogInvolved(Exception ex)
+        {
+            var isAuditLogException = false;
+
+            do      //traverse the nested exception chain to look for AuditLogException
+            {
+                if (ex.GetType() == typeof(AuditLogException))
+                {
+                    isAuditLogException = true;
```

Comment:
You should just "return true" here. There is no reason to keep walking up the hierarchy since it won't change the result.